Server/src/index.ts

```diff
@@ -3,169 +3,319 @@
-        this._modules = [
-            configurationModule,
-            evdriverModule,
-            monitoringModule,
-            reportingModule,
-            transactionsModule
-        ]
-        this._apis = [
-            new ConfigurationModuleApi(configurationModule, this._server, this._logger),
-            new EVDriverModuleApi(evdriverModule, this._server, this._logger),
-            new MonitoringModuleApi(monitoringModule, this._server, this._logger),
-            new ReportingModuleApi(reportingModule, this._server, this._logger),
-            new TransactionsModuleApi(transactionsModule, this._server, this._logger),
-        ];
-        if (this._config.modules.certificates) {
-           const certificatesModule = new CertificatesModule(this._config, this._cache, this._createSender(), this._createHandler(), this._logger)
-            this._modules.push(certificatesModule);
-            this._apis.push(new CertificatesModuleApi(certificatesModule, this._server, this._logger));
-        }
-        if (this._config.modules.smartcharging) {
-            const smartchargingModule = new SmartChargingModule(this._config, this._cache, this._createSender(), this._createHandler(), this._logger)
-            this._modules.push(smartchargingModule);
-            this._apis.push(new SmartChargingModuleApi(smartchargingModule, this._server, this._logger));
-        }
+    // Create Ajv JSON schema validator instance
+    this.initAjv(ajv);
 
-        process.on('SIGINT', this.shutdown.bind(this));
-        process.on('SIGTERM', this.shutdown.bind(this));
-        process.on('SIGQUIT', this.shutdown.bind(this));
-    }
+    // Initialize parent logger
+    this.initLogger();
 
-    protected _createSender(): IMessageSender {
-        return new RabbitMqSender(this._config, this._logger);
-    }
+    // Force sync database
+    this.forceDbSync();
+
+    // Set cache implementation
+    this.initCache(cache);
+
+    // Initialize Swagger if enabled
+    this.initSwagger();
+
+    // Register AJV for schema validation
+    this.registerAjv();
+
+    // Initialize module & API
+    // Always initialize API after SwaggerUI
+    this.initSystem(appName);
+
+    process.on('SIGINT', this.shutdown.bind(this))
+    process.on('SIGTERM', this.shutdown.bind(this))
+    process.on('SIGQUIT', this.shutdown.bind(this))
+  }
+
+  private initHealthCheck() {
+    this._server.get('/health', async () => {
+      return {status: 'healthy'}
+    })
+  }
+
+  private initAjv(ajv?: Ajv) {
+    this._ajv = ajv || new Ajv({
+      removeAdditional: 'all',
+      useDefaults: true,
+      coerceTypes: 'array',
+      strict: false
+    })
+    addFormats(this._ajv, {
+      mode: 'fast',
+      formats: ['date-time']
+    })
+  }
+
+  private initLogger() {
+    this._logger = new Logger<ILogObj>({
+      name: 'CitrineOS Logger',
+      minLevel: systemConfig.server.logLevel,
```

Comment:
i'm receiving this error when trying to get the docker running: 

`Server/src/index.ts(138,30): error TS2339:  Property 'server' does not exist on type '{ env: "development" | "production"; data:
 { sequelize: { host: string; port: number; database: string; username: string; password: string; storage: string; sync: boolean; dia
lect?: any; }; }; centralSystem: { ...; }; ... 4 more ...; maxCachingSeconds: number; }'`

i locally updated this to be `systemConfig.logLevel` and the docker containers ran successfully - should this same change be made to the PR as well or is this unique to my machine?

Reply:
@venus-yu great catch, this was something that I pulled from latest changes from Thana that was incorrectly merged, I resolved this